### DISCUSSION TITLE
ENSEMBLE_STAT_CLIMO_MEAN_USE_OBS not working?

### DISCUSSION THREAD

**Discussion by BinbinZhou-NOAA** [dtcenter/METplus#1902] [Configuration] [answered]
In EnsembleStat conf file for EC ensemble verification ,  the climo data are used. So I have following setting for climo
ENSEMBLE_STAT_CLIMO_MEAN_FILE_NAME = {clim_mean}
ENSEMBLE_STAT_CLIMO_MEAN_USE_OBS = True
ENSEMBLE_STAT_CLIMO_STDEV_FILE_NAME = {clim_stdev}
ENSEMBLE_STAT_CLIMO_STDEV_USE_OBS = True

The obs data are GRIB2 data while  the climo data  are GRIB1.  But this is not the reason since this setting works for GEFS. 
Not for sure the above settings just set  up the name and level of the fields ? 
or some others (center#, table# etc) are also set as same? 

Comparison (HGT P500):
EC analysis: 
SECTION 1:  98 0 2 1 1 2022 10 26 0 0 0 0 1
PRODUCT TEMPLATE 4. 0: ( PARAMETER = HGT      0 3 5 )  3 5 2 0 153 0 0 1 0 100 0 50000 255 0 0
GFS Analysis: 
SECTION 1:  7 0 2 1 1 2022 10 26 0 0 0 0 1
PRODUCT TEMPLATE 4. 0: ( PARAMETER = HGT      0 3 5 )  3 5 2 0 81 0 0 1 0 100 0 50000 255 0 0

The difference is in center id (7 ~ 98), and model id (81 ~ 153)

If  using another way:  explicitly set (there are 10 fields to be verified)
ENSEMBLE_STAT_CLIMO_MEAN_FIELD = { name="HGT"; level="P500"; }, { name="UGRD"; level="P850"; }, { name="VGRD"; level="P850"; }, { name="TMP"; level="P850"; }, { name="UGRD"; level="Z10"; }, { name="VGRD"; level="Z10"; }, { name="PRMSL"; level="L0"; }, { name="HGT"; level="P700"; }, { name="HGT"; level="P1000"; }, { name="TMP"; level="P500"; }
But the results show  4 fields are fine (correct CRPSS, decreasing with fhr) and other 6 fields  are not (wrong CRPSS, always ~ 0.99).

**Comment by JohnHalleyGotway** [comment, marked as answer]
@BinbinZhou-NOAA, thanks for following up via email with additional information about where/how to debug this. @jprestop and I have a meeting scheduled for this Thursday, 11/3/22, to look more closely at this.

**Reply by jprestop** [reply]
@BinbinZhou-NOAA 
In this file, /lfs/h2/emc/ptmp/binbin.zhou/evs/tmpnwprd/save.ecme_grid2grid_stats/conf/EnsembleStat_fcstECME_obsModelAnalysis_climoERA5.conf, you have the following set:
```
ENSEMBLE_STAT_CLIMO_MEAN_FIELD =

ENSEMBLE_STAT_CLIMO_MEAN_USE_OBS = True
```
Please see the documentation for [ENSEMBLE_STAT_CLIMO_MEAN_USE_OBS](https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html#term-ENSEMBLE_STAT_CLIMO_MEAN_USE_OBS) and note that the ENSEMBLE_STAT_CLIMO_MEAN_FIELD setting overrides the ENSEMBLE_STAT_CLIMO_MEAN_USE_OBS.  By leaving this field uncommented and blank you are setting that value to an empty list.

Please try removing  ENSEMBLE_STAT_CLIMO_MEAN_FIELD and rerunning.

We couldn't find any CRPSS values equal to .99 and would need more information to debug.

**Comment by BinbinZhou-NOAA** [comment]
John,

   After removing  ENSEMBLE_STAT_CLIMO_MEAN_FIELD=, the running is
successful to generate correct CRPSS
in the ECNT line type! The old one is "NA", not 0.99. Sorry for this typo.
Now I know that a setting placed ahead can overwrite  another setting
placed after in a conf file.

This case can be closed.

Thanks!

Binbin


On Thu, Nov 3, 2022 at 1:27 PM jprestop ***@***.***> wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA>
> In this file,
> /lfs/h2/emc/ptmp/binbin.zhou/evs/tmpnwprd/save.ecme_grid2grid_stats/conf/EnsembleStat_fcstECME_obsModelAnalysis_climoERA5.conf,
> you have the following set:
>
> ENSEMBLE_STAT_CLIMO_MEAN_FIELD =
>
> ENSEMBLE_STAT_CLIMO_MEAN_USE_OBS = True
>
> Please see the documentation for ENSEMBLE_STAT_CLIMO_MEAN_USE_OBS
> <https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html#term-ENSEMBLE_STAT_CLIMO_MEAN_USE_OBS>
> and note that the ENSEMBLE_STAT_CLIMO_MEAN_FIELD setting overrides the
> ENSEMBLE_STAT_CLIMO_MEAN_USE_OBS. By leaving this field uncommented and
> blank you are setting that value to an empty list.
>
> Please try removing ENSEMBLE_STAT_CLIMO_MEAN_FIELD and rerunning.
>
> We couldn't find any CRPSS values equal to .99 and would need more
> information to debug.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplus/discussions/1902#discussioncomment-4049677>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFOKRIUNIO766EZNWKLWGPYW5ANCNFSM6AAAAAARRPWH3A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

**Reply by jprestop** [reply]
Great @BinbinZhou-NOAA!  I'm glad it worked.  Thank you for letting us know.  I will go ahead and mark this Discussion as answered and lock it to prevent future posts.  Please feel free to start a new Discussion with any new issues.
